engines/dfc_provider/app/controllers/dfc_provider/api/base_controller.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+# Controller used to provide the API products for the DFC application
+module DfcProvider
+  module Api
+    class BaseController < ActionController::Base
+      rescue_from ActiveRecord::RecordNotFound, with: :not_found
+
+      before_action :check_authorization,
+                    :check_user,
+                    :check_enterprise
```

Comment:
Looking at these methods, I'd probably rename this last one to `#set_enterprise`. The other two are checking some validity, this method is setting the `@current_enterprise` instance variable.